var/spack/repos/builtin/packages/superlu-dist/package.py

```diff
@@ -139,38 +139,11 @@ def cache_test_sources(self):
         self.cache_extra_test_sources([self.examples_src_dir, self.mk_hdr])
 
     def test(self):
-        mk_file = join_path(self.install_test_root, self.mk_hdr)
-        # Replace 'SRC' with 'lib' in the library's path
-        filter_file(r'^(DSUPERLULIB.+)SRC(.+)', '\\1lib\\2', mk_file)
-        # Set library flags for all libraries superlu-dist depends on
-        filter_file(r'^LIBS.+\+=.+', '', mk_file)
-        filter_file(r'^LIBS[^\+]+=.+', 'LIBS = $(DSUPERLULIB)' +
-                    ' {0}'.format(self.spec['blas'].libs.ld_flags) +
-                    ' {0}'.format(self.spec['lapack'].libs.ld_flags) +
-                    ' {0}'.format(self.spec['parmetis'].libs.ld_flags) +
-                    ' {0}'.format(self.spec['metis'].libs.ld_flags) +
-                    ' $(CUDALIBS)',
-                    mk_file)
-        cuda_lib_opts = ''
-        if '+cuda' in self.spec:
-            cuda_lib_opts = ',-rpath,{0}'.format(
-                            self.spec['cuda'].libs.directories[0])
-        # Set the rpath for all the libs
-        filter_file(r'^LOADOPTS.+', 'LOADOPTS = -Wl' +
-                    ',-rpath,{0}'.format(self.prefix.lib) +
-                    ',-rpath,{0}'.format(self.spec['blas'].prefix.lib) +
-                    ',-rpath,{0}'.format(self.spec['lapack'].prefix.lib) +
-                    ',-rpath,{0}'.format(self.spec['parmetis'].prefix.lib) +
-                    ',-rpath,{0}'.format(self.spec['metis'].prefix.lib) +
-                    cuda_lib_opts,
-                    mk_file)
-
         test_dir = join_path(self.install_test_root, self.examples_src_dir)
-        test_exe = 'pddrive'
+        superludriver = join_path(self.prefix.lib, 'EXAMPLE', 'pddrive')
```

Comment:
Is `pddrive` automatically built and copied to that subdirectory of the install lib directory as part of the build process?   If so, the method should check for the version and output a `skipping` message that there are no tests for versions before it.  The `cache_test_sources` method should also be removed.

If `pddrive` is not automatically built, then removing the above code removes the process that builds it.

Reply:
That's a good point. Let me add the version checking, as pddrive is automatically built after a certain version.

Reply:
@tldahlgren Can you take a look at this again? I removed unused lines and added version detection in the test.